src/rgw/rgw_rados.cc

```diff
@@ -10053,12 +10067,19 @@ int RGWRados::Object::Read::read(int64_t ofs, int64_t end, bufferlist& bl)
         pbl = &read_bl;
       }
     }
+  } else {
+    store->open_pool_ctx(read_obj.pool, tail_io_ctx);
+    tail_io_ctx.locator_set_key(read_obj.loc);
   }
 
   ldout(cct, 20) << "rados->read obj-ofs=" << ofs << " read_ofs=" << read_ofs << " read_len=" << read_len << dendl;
   op.read(read_ofs, read_len, pbl, NULL);
 
-  r = state.io_ctx.operate(read_obj.oid, &op, NULL);
+  if (reading_from_head)
```

Comment:
@fangyuxiangGL why do we need a special handling here? The code already knows how to deal with different head and tail pools. Everyting should be in the manifest. Maybe you missed updating the manifest correctly?